VKapp/VKapp/Controllers/OneFriendWithSwipeController.swift

```diff
@@ -0,0 +1,101 @@
+// OneFriendWithSwipeController.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// List of photos one friend with swipe
+final class OneFriendWithSwipeController: UIViewController {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let xTranslation: CGFloat = 500
+    }
+
+    // MARK: - IBOutlets
+
+    @IBOutlet private var photoImageView: UIImageView!
+
+    // MARK: - Public Properties
+
+    var photos: [UIImage?] = []
+
+    // MARK: Private Properties
+
+    private var index = Int()
+
+    // MARK: - Life Cycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configUI()
+        createSwipeRecognizer()
+    }
+
+    // MARK: Private Methods
+
+    private func configUI() {
+        guard let image = photos.first else { return }
+        photoImageView.image = image
+    }
+
+    private func createSwipeRecognizer() {
+        photoImageView.isUserInteractionEnabled = true
+        let swipeRight = UISwipeGestureRecognizer(target: self, action: #selector(doSwipeAction(gesture:)))
+        swipeRight.direction = .right
+        photoImageView.addGestureRecognizer(swipeRight)
+
+        let swipeLeft = UISwipeGestureRecognizer(target: self, action: #selector(doSwipeAction(gesture:)))
+        swipeLeft.direction = .left
+        photoImageView.addGestureRecognizer(swipeLeft)
+
+        let swipeDown = UISwipeGestureRecognizer(target: self, action: #selector(doSwipeAction(gesture:)))
+        swipeDown.direction = .down
+        photoImageView.addGestureRecognizer(swipeDown)
+    }
+
+    private func swipe(translationX: CGFloat, differenceIndex: Int, rotatingAngle: CGFloat) {
+        index += differenceIndex
+        guard index < photos.count, index >= 0 else {
+            index -= differenceIndex
+            return
+        }
+        UIView.animate(withDuration: 0.5) {
+            let translation = CGAffineTransform(translationX: translationX, y: 0)
+            let scale = CGAffineTransform(scaleX: 0.7, y: 0.7)
+            let rotating = CGAffineTransform(rotationAngle: rotatingAngle)
+            self.photoImageView.transform = translation.concatenating(scale).concatenating(rotating)
+            self.photoImageView.layer.opacity = 0.5
+        } completion: { _ in
+            let newPhotoScale = CGAffineTransform(scaleX: 0.5, y: 0.5)
+            self.photoImageView.transform = CGAffineTransform(translationX: -translationX, y: 0)
+                .concatenating(newPhotoScale)
+            UIView.animate(withDuration: 0.5) {
+                self.photoImageView.layer.opacity = 1
+                self.photoImageView.transform = .identity
+                self.photoImageView.image = self.photos[self.index]
```

Comment:
```suggestion
        UIView.animate(withDuration: ) {
            let translation = CGAffineTransform(translationX: translationX, y: )
            let scale = CGAffineTransform(scaleX:  y: )
            let rotating = CGAffineTransform(rotationAngle: rotatingAngle)
            self.photoImageView.transform = translation.concatenating(scale).concatenating(rotating)
            self.photoImageView.layer.opacity = 
        } completion: { _ in
            let newPhotoScale = CGAffineTransform(scaleX: , y: )
            self.photoImageView.transform = CGAffineTransform(translationX: -translationX, y: )
                .concatenating(newPhotoScale)
            UIView.animate(withDuration: ) {
                self.photoImageView.layer.opacity = 
                self.photoImageView.transform = .identity
                self.photoImageView.image = self.photos[self.index]
```
числа и коэффиуиенты выносим в константы
статья про "магические числа"